Modules/pyexpat.c

```diff
@@ -1803,17 +1806,17 @@ add_errors_module(PyObject *mod)
         goto error;
     }
 
-    if (PyModule_AddObject(errors_module, "codes", Py_NewRef(codes_dict)) < 0) {
-        Py_DECREF(codes_dict);
+    int rc = PyModule_AddObjectRef(errors_module, "codes", codes_dict);
+    Py_CLEAR(codes_dict);
```

Comment:
I think that you are changing the previous behaviour here. Right now `Py_CLEAR(codes_dict);` would be called in all cases, previously it was only called on success. Is it intentional?

Reply:
Since `codes_dict` is a local variable that holds a strong reference, we must ensure that we decref `codes_dict` on all exit paths. For the happy path, the code will look like this (as of this PR, error handling removed):

```c
    PyModule_AddObjectRef(errors_module, "codes", codes_dict);
    Py_CLEAR(codes_dict);
    PyModule_AddObjectRef(errors_module, "messages", rev_codes_dict);
    Py_CLEAR(rev_codes_dict);
    return 0;
```

If the first `PyModule_AddObjectRef` fails, the path will look like this:

```c
    int rc = PyModule_AddObjectRef(errors_module, "codes", codes_dict);
    Py_CLEAR(codes_dict);
    if (rc < 0) {
        goto error;
    }

    /* ... */

error:
    Py_XDECREF(codes_dict);
    Py_XDECREF(rev_codes_dict);
    return -1;
```

So, we need `Py_CLEAR` to assign `NULL` to `codes_dict` so we don't end up decref'ing `codes_dict` twice in the error handler.

There should be no change in behaviour.

Reply:
Alternatively, we could rewrite the code at the `error:` label, but that would be more churn IMO.